app/src/test/java/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModelTest.kt

```diff
@@ -620,4 +626,102 @@ class ConnectionsListViewModelTest : ViewModelTest() {
             )
         )
     }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase1() {
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_NEGATIVE,
+            actionResId = R.string.actions_proceed
+        )
+
+        assertNull(viewModel.onAskPermissionsEvent.value)
+
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_POSITIVE,
+            actionResId = R.string.actions_proceed
+        )
+
+        assertNotNull(viewModel.onAskPermissionsEvent.value)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase2() {
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_NEGATIVE,
+            actionResId = R.string.actions_go_to_settings
+        )
+
+        assertNull(viewModel.onGoToSettingsEvent.value)
+
+        viewModel.onDialogActionIdClick(
+            dialogActionId = DialogInterface.BUTTON_POSITIVE,
+            actionResId = R.string.actions_go_to_settings
+        )
+
+        assertNotNull(viewModel.onGoToSettingsEvent.value)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun onDialogActionIdClickTestCase3() {
+        //given
```

Comment:
all test cases should have given-when-then annotation, not only onDialogActionIdClickTestCase3 and onDialogActionIdClickTestCase4